src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,200 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+import javafx.util.Pair;
+
+import java.util.HashMap;
+
+public class Parser {
+    private static final BadCommandException INSUFFICIENT_PARAMS_ERROR =
+            new BadCommandException("There are insufficient or invalid parameters!");
+    private static final BadCommandException UNKNOWN_COMMAND_ERROR =
+            new BadCommandException("I'm sorry, but I don't know what that means :-(");
+    private static final String PARAMS_DELIMITER =  "/";
+
+    private enum Command {
+        LIST("list"),
+        MARK("mark"),
+        UNMARK("unmark"),
+        DELETE("delete"),
+        TODO("todo"),
+        DEADLINE("deadline"),
+        EVENT("event"),
+        FIND("find");
+        private final String commandStr;
+        private static final HashMap<String, Command> STRING_TO_COMMAND_MAP = new HashMap<>();
+        static {
+            for (Command command: values()) {
+                STRING_TO_COMMAND_MAP.put(command.commandStr, command);
+            }
+        }
```

Comment:
I notice that this class feels a bit long when I am reading it, maybe can separate the enum Command into a new class.